client/app/components/Link.tsx

```diff
@@ -15,10 +21,36 @@ function DefaultLinkComponent({ children, ...props }: React.AnchorHTMLAttributes
 
 Link.Component = DefaultLinkComponent;
 
-function Link({ tabIndex = 0, ...props }: LinkProps) {
-  return <Link.Component tabIndex={tabIndex} {...props} />;
+function Link({ tabIndex = 0, children, ...props }: LinkProps) {
+  return (
+    <Link.Component tabIndex={tabIndex} {...props}>
+      {children}
+    </Link.Component>
+  );
 }
 
+function LinkWithIcon({ icon, alt, children, ...props }: LinkWithIconProps) {
+  return (
+    <Link {...props}>
+      {children} {icon} <span className="sr-only">{alt}</span>
+    </Link>
+  );
+}
+
+Link.WithIcon = LinkWithIcon;
+
+function ExternalLink({
+  icon = <i className="fa fa-external-link" aria-hidden="true" />,
+  alt = "(opens in a new tab)",
+  ...props
+}: LinkWithIconProps) {
+  const externalLinkProps = { target: "_blank", rel: "noopener noreferrer", icon, alt };
```

Comment:
Why not just use these props directly in the next line?

Reply:
Not doing so allows the user to overhead them via normal props, which is intended (except for target, which is banned from `<Link>`

Reply:
IMHO, considering that all props here are constants (or other props), inline version will look more clear:

```jsx
return <Link.WithIcon target="_blank" rel="noopener noreferrer" icon={icon} alt={alt} {...props} />;
```

Using object for additional props usually makes sense if you need some conditional or computed props - in this case it will make code cleaner. But in this case I don't see actual need for that